File: WorldFeed/Server/src/WorldFeed.Common/Public/Models/Interfaces/IWebhook.cs
namespace WorldFeed.Common.Public.Models.Interfaces
{
    using System;

    using WorldFeed.Common.Public.Models.Interfaces.DTO.Webhooks;

    public interface IWebhook
    {
        IWebhookDTO WebhookDTO { get; }

        /// <summary>
        /// Webhook identifier
        /// </summary>
        string Id { get; }

        /// <summary>
        /// Registered url
        /// </summary>
        string Url { get; }

        /// <summary>
        /// Whether the webhook succeeded its last crc challenge
        /// </summary>
        bool Valid { get; }

        /// <summary>
        /// Registration date
        /// </summary>
        DateTime CreatedAt { get; }

        /// <summary>
        /// Registered Uri
        /// </summary>
        Uri Uri { get; }
    }
}

File: Sharebook/Client/src/app/core/Public/Models/Interfaces/IWebhook.ts
import {IWebhookDTO} from "./DTO/Webhooks/IWebhookDTO";
import DateTime from "typescript-dotnet-commonjs/System/Time/DateTime";
import Uri from "typescript-dotnet-commonjs/System/Uri/Uri";

export interface IWebhook {
  webhookDTO: IWebhookDTO;

  // Webhook identifier
  id: string;

  // Registered url
  url: string;

  // Whether the webhook succeeded its last crc challenge
  valid: boolean;

  // Registration date
  createdAt: DateTime; // DateTimeOffset;

  // Registered Uri
  uri: Uri;
}
